sys/cam/ctl/ctl_frontend_ioctl.c

```diff
@@ -367,7 +369,7 @@ ctl_ioctl_do_datamove(struct ctl_scsiio *ctsio)
 		ext_sglist = (struct ctl_sg_entry *)malloc(ext_sglen, M_CTL,
 							   M_WAITOK);
 		ext_sglist_malloced = 1;
-		if (copyin(ctsio->ext_data_ptr, ext_sglist, ext_sglen) != 0) {
+		if (copyin(ctsio->ext_data_uptr, ext_sglist, ext_sglen) != 0) {
```

Comment:
It is unclear to me what sets `ext_data_uptr`, do we need a corresponding `__USER_CAP` conversion from it somewhere, or we assume that this will not work unless the userspace is purecap?

Reply:
Userland code that sets `ext_data_ptr` actually sets `ext_data_uptr`.  The union only exists in the kernel, and in purecap userland `ext_data_ptr` is a capability.

Reply:
Makes sense now, thanks